hw2/02-search.js

```diff
@@ -0,0 +1,58 @@
+const userInput = document.querySelector("#getData");
+userInput.addEventListener("click", handleClick); // grab data on input click
+
+// Search character array for a match and return the results
+function handleClick() {
+  const search = document.getElementById("userInput").value.trim();
+  console.log(search);
+
+  // Create array of results
+  const results = characters.filter((toFind) =>
+    toFind.name.toLocaleLowerCase().includes(search.toLocaleLowerCase())
+  );
+
+  console.log(results);
+
+  // Output the results
+  displayCharacterComponents(results);
+}
+
+// Build a character object
+function createCharacterComponent(characterData) {
+  const container = document.createElement("section");
+  container.classList.add("card", "mb-3");
+
+  const cardBody = document.createElement("div");
+  cardBody.classList.add("card-body");
+
+  let name = document.createElement("h3");
+  name.classList.add("card-title");
+  name.textContent = characterData.name;
+
+  let birthYear = document.createElement("p");
+  birthYear.classList.add("card-text");
+  birthYear.textContent = `Birth year: ${characterData.birth_year}`;
+
+  cardBody.appendChild(name);
+  cardBody.appendChild(birthYear);
+  container.appendChild(cardBody);
+  return container;
+}
+
+// Display results to DOM
+function displayCharacterComponents(resultsArray) {
+  const display = document.getElementById("results");
+
+  // Check for null
+  if (!resultsArray || resultsArray.length === 0) {
+    console.error("The array is empty.");
+    return;
+  }
+
+  // Display results
+  resultsArray.forEach((element) => {
+    console.log(element.name, element.birth_year);
+    const charComponent = createCharacterComponent(element);
```

Comment:
Having this call nested within took me a moment to wrap my head around. It likely wouldn't hurt to combine the functions or call the functions separately from the handleClick function. That being said, this is somewhat more of a preference than a requirement

Reply:
Yeah.  It's basically going through the filtered results and creating JS objects with formatting for each element in the array.  I tried to abstract the two because building JS objects with formatting is verbose.